### PR TITLE
Expand on the errors guide for actix-web

### PR DESCRIPTION
**Pull request by brandur**
As discussed in a previous actix-web thread on errors [1], it might be
nice to add a little more flavor to the user guide on error handling,
with some focus on recommended practices.

This patch makes a few changes:

1. Adds a "Recommended practices in error handling" section with a few
   more ideas on how errors might be effectively used in a real
   application.
2. Adds a "Compatibility with failure" section.
3. Rewrites most of the other sections for clarity/succinctness.

[1] https://github.com/actix/actix-web/issues/197

cc @fafhrd91

### CODE REVIEW

File: actix-web/src/sec-5-errors.md [hunk 1]
```diff
@@ -135,19 +151,114 @@ fn index(req: HttpRequest) -> Result<&'static str> {
...
+See the [API documentation for actix-web's `error` module][errorhelpers] for a
+full list of available error helpers.
 
-## Error logging
+## Compatibility with failure
+
+Actix-web provides automatic compatibility with the [failure] library so that
+errors deriving `fail` will be converted automatically an actix error. Keep in
+that those errors will render with the default *500* status code unless you
+also provide your own `error_response()` implementation.
```
Comment (brandur) [inline actix-web/src/sec-5-errors.md:162]: Would you mind checking to see if this paragraph is correct? I still haven't really used failure, and I might have some bad ideas about it.

Comment (fafhrd91) [reply actix-web/src/sec-5-errors.md:162]: that is correct

**Review by fafhrd91 — APPROVED** [review]
looks great

**Comment by brandur** [comment]
Excellent. Thanks @fafhrd91.

Do you think you could add the `book` repository to the "Contributors" team (so that I can merge)? I think that it was accidentally omitted.

**Comment by fafhrd91** [comment]
Forgot about contributors. Added

**Comment by brandur** [comment]
Thanks again!

**Pull request merged**